packages/kamino-sdk/src/Kamino.ts

```diff
@@ -185,6 +197,106 @@ export class Kamino {
 
   getTokenMap = () => this._tokenMap;
 
+  getDepositableTokens = async (): Promise<CollateralInfo[]> => {
+    let config = await GlobalConfig.fetch(this._connection, this._globalConfig);
+    if (!config) {
+      throw Error(`Could not fetch globalConfig  with pubkey ${this.getGlobalConfig().toString()}`);
+    }
+    const collateralInfos = await this.getCollateralInfo(config.tokenInfos);
+
+    let depositableTokens: CollateralInfo[] = [];
+    collateralInfos.forEach((element) => {
+      if (element.mint.toString() != SystemProgram.programId.toString()) {
+        depositableTokens.push(element);
+      }
+    });
+
+    return depositableTokens;
+  };
+
+  getSupportedDexes = (): Dex[] => ['ORCA', 'RAYDIUM'];
+
+  // const whirlpoolConfig: PublicKey = new PublicKey("2LecshUwdy9xi7meFgHtFJQNSKk4KdTrcpvaB56dP2NQ");
+
+  // todo: see if we can read this dinamically
+  getFeeTiersForDex = (dex: Dex): Decimal[] => {
+    if (dex == 'ORCA') {
+      return [new Decimal(0.0001), new Decimal(0.0005), new Decimal(0.003), new Decimal(0.01)];
+    } else if (dex == 'RAYDIUM') {
+      return [new Decimal(0.0001), new Decimal(0.0005), new Decimal(0.0025), new Decimal(0.01)];
+    } else {
+      throw new Error(`Dex ${dex} is not supported`);
+    }
+  };
+
+  isPoolInitializedForDexPairTier = async (
+    dex: Dex,
+    poolTokenA: PublicKey,
+    poolTokenB: PublicKey,
+    fee: Decimal
+  ): Promise<boolean> => {
+    if (dex == 'ORCA') {
+      let orcaPools = await this.getOrcaPoolsForTokens(poolTokenA, poolTokenB);
+      orcaPools.forEach((element) => {
+        if (element.lpFeeRate == fee.toNumber()) {
+          return true;
+        }
+      });
+      return false;
+    } else if (dex == 'RAYDIUM') {
+      let raydiumPools = await this.getRaydiumPoolsForTokens(poolTokenA, poolTokenB);
+      raydiumPools.forEach((element) => {
+        if (element.ammConfig.tradeFeeRate == fee.toNumber()) {
+          return true;
+        }
+      });
+      return false;
+    } else {
+      throw new Error(`Dex ${dex} is not supported`);
+    }
+  };
+
+  getExistentPoolsForPair(dex: Dex, poolTokenA: PublicKey, poolTokenB: PublicKey) {
+    if (dex == 'ORCA') {
+      return this.getOrcaPoolsForTokens(poolTokenA, poolTokenB);
+    } else if (dex == 'RAYDIUM') {
+      return this.getRaydiumPoolsForTokens(poolTokenA, poolTokenB);
+    } else {
+      throw new Error(`Dex ${dex} is not supported`);
+    }
+  }
+
+  getOrcaPoolsForTokens = async (poolTokenA: PublicKey, poolTokenB: PublicKey): Promise<OrcaPool[]> => {
+    let pools: OrcaPool[] = [];
+    let whirlpools = await this._orcaService.getOrcaWhirlpools();
```

Comment:
out of curiosity, if you do this for mainnet, how many numbers does it return for each dex?

secondly, we should be doing smarter fetches with memcpy filters, we know how to do this and i assume will be a big speedup, but it's an implementation change, not an interface one, so we can do as followup if we notice slowness in the ui

Reply:
This actually makes an API request  to Orca/Raydium whirlpools endpoint they use in their UI as well. Doesn't make an rpc call

Reply:
oh nice, cool, if fast enough, not necessary, this is on strategy creation, not on every page load, not super necessary to optimize for speed initially imo

Reply:
Good point yeah, might be good enough in this case